cmd/snap/cmd_sign.go

```diff
@@ -38,6 +38,10 @@ of the assertion can be specified through a "body" pseudo-header.
 `)
 
 type cmdSign struct {
+	Positional struct {
+		Filename string
+	} `positional-args:"yes" required:"no"`
```

Comment:
hmm it's a bit weird, but this somehow is still triggering the tests to think that the Filename argument is required:

```
----------------------------------------------------------------------
FAIL: cmd_sign_test.go:43: SnapKeysSuite.TestHappyDefaultKey
cmd_sign_test.go:47:
    c.Assert(err, IsNil)
... value *flags.Error = &flags.Error{Type:0x7, Message:"the required argument `<filename>` was not provided"} ("the required argument `<filename>` was not provided")
----------------------------------------------------------------------
FAIL: cmd_sign_test.go:55: SnapKeysSuite.TestHappyNonDefaultKey
cmd_sign_test.go:59:
    c.Assert(err, IsNil)
... value *flags.Error = &flags.Error{Type:0x7, Message:"the required argument `<filename>` was not provided"} ("the required argument `<filename>` was not provided")
2020/01/06 22:40:30.945933 cmd_run.go:779: cannot extract runtime data: cannot parse end of exec profile: EOF
OOPS: 444 passed, 13 skipped, 2 FAILED
--- FAIL: Test (3.46s)
FAIL
FAIL	github.com/snapcore/snapd/cmd/snap	3.560s
```

Reply:
@anonymouse64 
yes, i test this case too late

Reply:
@anonymouse64 
It was late yesterday. 
Unfortunately I have not found a place where these reflection tags are interpreted. 

I start a thread here:
https://forum.snapcraft.io/t/snap-sign-should-the-support-for-stdin-data-be-deprecated-or-removed/14943

Reply:
it's a bit silly, but what you want is to remove the `required` altogether.

`required`'s presence on positional means you'll be specifying `required` per field inside the struct.

Reply:
(it might be more involved than that, this bit of go-flags is a bit crufty)